### DISCUSSION TITLE
Wireless passwords do not remain stored when reconnecting to a saved network

### DISCUSSION THREAD

**Discussion by omakuva** [Nitrux/nitrux-bug-tracker#136] [Q&A] [answered]
On Nitrux OS 2.9.0

In the system settings module, connections (ID and PW) are not stored. 
By restarting the device, previously accessed connections are not enlisted. – So, credentials need be re-entered anew.

tbc.

If the issue concerns hardware, use `hw-probe`, run it using the following command `sudo -E hw-probe -all -upload`, and add the URL link.

**Comment by UriHerrera** [comment]
NetworkManager saves the passwords for wireless connections in this path, `/etc/NetworkManager/system-connections`, when the network connection is enabled for all users; see the image below.

![imagen](https://github.com/Nitrux/nitrux-bug-tracker/assets/3053525/bf268405-09dd-4bdb-9ebe-cc2bb5545a8b)

- Can you verify if there are any files in that directory?
- Did you use the same wireless connection once you've booted to the installed system that you used to do the installation?

The option "All users may connect to this network" is the default selection; if you uncheck this option, then `plasma-nm` will (presumably) attempt to use KWallet Manager, which we do not include, to create a keyring to save the password for the wireless connection for the user.

Since NetworkManager saves files to `/etc` if the network connection is different than the one used during installation, which Calamares will automatically save as indicated by our settings, then adding a new connection will result in the password not being saved because the directory `/etc` is part of the root directory, and the root is immutable.

**Comment by omakuva** [comment]
- [ **yes**]     Can you verify if there are any files in that directory? The one I currently use, and the file of the connection used during the installation-procedure.

- [ **not necessarily**]     Did you use the same wireless connection once you've booted to the installed system that you used to do the installation? All 'new' shall be added by hand, every single time. – 

I guess, the reason is – as you remarked – that the root directory cannot be modified this way.

Also, please check : https://duckduckgo.com/?q=kde+does+not+save+desktop+setting&t=ffab&atb=v386-1&ia=web – possibly related. – 


(P.S.: Now, I've similar network config related issues, for e.g. if I follow " https://nxos.org/tutorial/filesystem-security-privacy-and-anonymization-features-in-nitrux/#online-anonymization-using-tor-and-dnscrypt-proxy-2 " settings (socks5 host and port, and IPv4 and IPv6 settings ) are not stored. – Perhaps, we shall open a new report about this.)

**Comment by UriHerrera** [comment]
> [ yes] Can you verify if there are any files in that directory? The one I currently use, and the file of the connection used during the installation-procedure.

The password for the new network connection you add does not seem to persist after rebooting, yes? i.e., if you add it, the file is not found in that path after rebooting.

> [ not necessarily] Did you use the same wireless connection once you've booted to the installed system that you used to do the installation? All 'new' shall be added by hand, every single time. –

Is that a yes or a no? I presume it's a no.

I want to mention that I have noticed that it's also a matter of the network itself. For example, I reinstalled the distribution twice on another computer I have. This computer can only connect to a network using a USB wireless dongle. 

TL;DR, I used two networks from two different ISPs with different routers. One network connection never asks for the password upon rebooting the installed system despite KWallet Manager not being present; the other network connection constantly asks for the password.

> (P.S.: Now, I've similar network config related issues, for e.g. if I follow " https://nxos.org/tutorial/filesystem-security-privacy-and-anonymization-features-in-nitrux/#online-anonymization-using-tor-and-dnscrypt-proxy-2 " settings (socks5 host and port, and IPv4 and IPv6 settings ) are not stored. – Perhaps, we shall open a new report about this.)

The [tutorial](https://nxos.org/tutorial/filesystem-security-privacy-and-anonymization-features-in-nitrux/#online-anonymization-using-tor-and-dnscrypt-proxy-2) mentions the following. 

> ⚠️ Important: The commands below create, copy, edit, or move files in the root directory; _see [XFS Features and Root Immutability in Nitrux](https://nxos.org/tutorial/filesystem-security-privacy-and-anonymization-features-in-nitrux/#xfs-features-and-root-immutability-in-nitrux)_.

Running the commands as is will write to the upper directory in the overlay, not the lower directory. That's not a bug; that's how OverlayFS and, thus, that is how overlayroot works. To make the changes permanent, use  `overlayroot-chroot` as the tutorial indicates.

Or were you referring to the Plasma configuration?

Could you let me know if you are getting any notifications when you add the network or the password?

**Comment by UriHerrera** [comment]
[KWallet Manager was added in Nitrux 3.0.0](https://nxos.org/changelog/release-announcement-nitrux-3-0-0/). This addition should allow users to store passwords for wireless connections in a user-accessible directory, i.e., their home directory.

- 🔰 **Information**: To use KWalletManager with GPG keys, we recommend creating a key using KGpg. To do this, see our tutorial on [_How to use KWalletManager in Nitrux_](https://nxos.org/tutorial/how-to-use-kwalletmanager-in-nitrux/).

----

However, the option  "All users may connect to this network" remains the default selection as it comes from Plasma, and I do not know if it can be modified or, if it can, what file should be modified.

----

I've changed the label of this issue to upstream-bug since, from our perspective, this is a problem caused by a default option in upstream, as was the case with casper (see #119), where we had to apply a workaround for another issue.

Therefore, adding `/etc` to the partition layout as a separate partition will be a workaround for this problem.

**Comment by UriHerrera** [comment]
## Update

I've tested adding `/etc` as a separate partition on our partition layout in Calamares; however, this caused a problem; doing this prevented the distribution from completing the boot process as OpenRC could not find the configuration for the runlevels.

So that's not a viable solution.

**Reply by UriHerrera** [reply, marked as answer]
Solved in https://github.com/Nitrux/iso-tool/commit/4ea3b853795fe6d2e52a6ccd2fcd14979049544e

Check the PSA for the latest release to apply it to an existing installation.

https://nxos.org/psa/psa-nitrux-3-1-0/
